sherpa/astro/data.py

```diff
@@ -1807,6 +1856,149 @@ def counts(self):
     def counts(self, val):
         self.y = val
 
+    # Override the mask handling because the mask matches the grouped
+    # data length, not the independent axis.
+    #
+    @Data1D.mask.setter
+    def mask(self, val):
+
+        # We only need to over-ride the behavior if the data is
+        # grouped and val is a sequence (so we test with isscalar
+        # rather than iterable, to avoid selecting strings).
+        #
+        if self.grouped and val is not None and not numpy.isscalar(val):
+            # The assumption is that is the data is grouped then it contains data.
```

Comment:
And now for the important suggestions...
```suggestion
            # The assumption is that if the data is grouped then it contains data.
```